velox/substrait/SubstraitToVeloxPlan.h

```diff
@@ -92,6 +98,9 @@ class SubstraitVeloxPlanConverter {
   /// The lengths to be scanned.
   std::vector<u_int64_t> lengths_;
 
+  /// The rowVector of the velox ValuesNode to be read
+  std::vector<RowVectorPtr> vectors_;
```

Comment:
Why does this need to be a member variable? Looks like this can be just a local variable in the method that converts ReadRel_VirtualTable to ValuesNode.

Reply:
we will need this if we do the test  for only substrait2Velox convertor

Reply:
I am planning to do the substrait2Velox test like below,  the `testInputs` is the velox plan that we convert from the substrait json file.  That's the place where we will need `getVectors()` . What  do you think?
```
  void testProject(std::shared_ptr<doTest> testInputs) {
    auto vectors = testInputs->getVectors();
    std::cout<<"vectors.size()"<<vectors.size()<<std::endl;
    createDuckDbTable(vectors);
    assertQuery(testInputs->getVPlan(), "SELECT c0, c1 FROM tmp WHERE c1 > 0");
  }
```

Reply:
I'm assuming the test flow is as follows. Please, correct me if that's not the case.

1. Read JSON file containing Substrait plan. Parse into Substrait plan object.
2. Convert Substrait plan object into Velox plan object. In this case it will contain ValuesNode which contains std::vector<RowVectoPtr> inside.
3. Run Velox query and verify results.

The problem will arise with "verify results as there is no SQL that matches the query plan to run on DuckDB. If you limit the test to simple plans with just one ValuesNode, then you can take Velox plan, extract ValuesNode, extract vectors using ValuesNode::values() getter, write them into DuckDB table, then use SELECT * FROM t query against DuckDB. 

This way you don't need to store vectors in the converter.

Does this make sense?

Reply:
Perhaps, this can be generalized by extracting all Values nodes from the query plan and creating corresponding DuckDB tables.

Reply:
got it, will try to get the vectors from the ValuesNode in  velox Plan that we converter from substrait plan, not using the getVectors(). am I right?

Reply:
why  the problem will arise with "verify results as there is no SQL that matches the query plan to run on DuckDB" ? it's the reason that we use c1 not tmp.c1 ? eg, assertQuery(testInputs->getVPlan(), "SELECT tmp.c0, tmp.c1 FROM tmp WHERE tmp.c1 > 0");

Reply:
> got it, will try to get the vectors from the ValuesNode in velox Plan that we converter from substrait plan, not using the getVectors(). am I right?

That's right. 

> why the problem will arise with "verify results as there is no SQL that matches the query plan to run on DuckDB" ?

I got confused. Like you are pointing out things should work just fine.

Reply:
so that it's ok that we use the getVectors() ?

Reply:
Yes. We can lookup ValuesNode and call ValuesNode::getVectors().